File: modules/x-engine-module-localstorage/android/module-localstorage/src/main/java/com/zkty/modules/loaded/jsapi/__xengine__module_localstorage.java
package com.zkty.modules.loaded.jsapi;

import android.content.Context;
import android.util.Log;

import com.tencent.mmkv.MMKV;
import com.zkty.modules.engine.core.IApplicationListener;
import com.zkty.modules.engine.utils.ActivityUtils;
import com.zkty.modules.loaded.util.SharePreferenceUtils;

import com.zkty.modules.dsbridge.CompletionHandler;

import java.io.File;

public class __xengine__module_localstorage extends xengine__module_localstorage implements IApplicationListener {

    @Override
    public void onAppCreate(Context context) {
        String dir = new File("/data/data/" + context.getPackageName() + "/shared_prefs").getAbsolutePath() + "/mmkv";
        String rootDir = MMKV.initialize(dir);
        Log.d("localstorage", "mmkv init root: " + rootDir);
    }

    @Override
    public void onAppLowMemory() {

    }

    @Override
    public void _set(StorageSetDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        SharePreferenceUtils.put(ActivityUtils.getCurrentActivity(), dto.isPublic, dto.key, dto.value);
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = "success";
        handler.complete(statusDTO);
    }

    @Override
    public void _get(StorageGetDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        String result = (String) SharePreferenceUtils.get(ActivityUtils.getCurrentActivity(), dto.isPublic, dto.key, "");
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = result;
        handler.complete(statusDTO);
    }

    @Override
    public void _remove(StorageRemoveDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        SharePreferenceUtils.remove(ActivityUtils.getCurrentActivity(), dto.isPublic, dto.key);
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = "success";
        handler.complete(statusDTO);
    }

    @Override
    public void _removeAll(StorageRemoveAllDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        SharePreferenceUtils.clear(ActivityUtils.getCurrentActivity(), dto.isPublic);
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = "success";
        handler.complete(statusDTO);
    }


    @Override
    public String moduleId() {
        return super.moduleId();
    }
}

File: modules/x-engine-jsi-legacy/iOS/Class/localstorage/__xengine__module_localStorage.m
//
//  __xengine__module_localstorage.m
//  AFNetworking
//
//  Created by Chenwuzheng on 2020/8/6.
//

#import "__xengine__module_localStorage.h"
#import "JSIContext.h"

#import "XEDataSaveManage.h"

@implementation __xengine__module_localstorage
JSI_MODULE(__xengine__module_localstorage)


- (void)_get:(StorageGetDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = [[XEDataSaveManage instance] getLocalStorage:dto.key withIsPublic:dto.isPublic];
    completionHandler(ret , YES);
}

- (void)_remove:(StorageRemoveDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    [[XEDataSaveManage instance] removeLocalStorageItem:dto.key withIsPublic:dto.isPublic];
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = @"success";
    completionHandler(ret , YES);
}

- (void)_removeAll:(StorageRemoveAllDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    [[XEDataSaveManage instance] removeLocalStorageAll:dto.isPublic];
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = @"success";
    completionHandler(ret , YES);
}

- (void)_set:(StorageSetDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    [[XEDataSaveManage instance] setLocalStorage:dto.key withValue:dto.value withIsPublic:dto.isPublic];
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = @"success";
    completionHandler(ret , YES);
}

@end
